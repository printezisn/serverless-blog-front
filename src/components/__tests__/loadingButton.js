import React from "react"
import renderer from "react-test-renderer"

import LoadingButton from "../loadingButton"

describe("LoadingButton", () => {
    it("renders correctly when it's not loading", () => {
        const button = renderer
            .create(
                <LoadingButton
                    type="submit"
                    className="button"
                    isLoading={false}
                >
                    Test Text
                </LoadingButton>
            )
            .toJSON()
        expect(button).toMatchSnapshot()
    })

    it("renders correctly when it's loading", () => {
        const button = renderer
            .create(
                <LoadingButton
                    type="submit"
                    className="button"
                    isLoading={true}
                >
                    Test Text
                </LoadingButton>
            )
            .toJSON()
        expect(button).toMatchSnapshot()
    })
})
