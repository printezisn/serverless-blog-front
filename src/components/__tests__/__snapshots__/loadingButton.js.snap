// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton renders correctly when it's loading 1`] = `
<button
  className="button is-loading"
  disabled={true}
  type="submit"
>
  Test Text
</button>
`;

exports[`LoadingButton renders correctly when it's not loading 1`] = `
<button
  className="button"
  type="submit"
>
  Test Text
</button>
`;
